### PR TITLE
Pack metadata template creation

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
In Progress

## Related Issues
related to: https://github.com/demisto/etc/issues/19786

## Description
Added metadata template and option for interactive metadata fill.

### CODE REVIEW

**Comment by IkaDemisto** [comment]
@anara123 @yaakovi please review and comment regarding the metadata template (https://github.com/demisto/demisto-sdk/blob/a8776fc6e78acb637b94e0b64ff9be4113e10757/demisto_sdk/common/templates/PackData/metadata_template.json).

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "Display name of the pack:",
+  "description": "Description of the pack:",
+  "support": "Support of the pack, options are demisto / partner / developer / community :",
+  "beta": "Whether is beta version, options are True or False :",
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: set this to be default false, and no need to ask the user

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "deprecated": "Whether pack is deprecated, options are True or False:",
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: set this to false and no need to ask the user

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "certification": "Certification of the pack, options are certified or verified :",
+  "serverMinVersion": "Server min version:",
+  "serverLicense": "Server license, options are standard or premium :",
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: remove this , there is no standard or premium anymore

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "currentVersion": "Current version of the pack:",
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: set this to 1.0.0 by default

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "author": "Author of the pack:",
+  "url": "The url of support, should represent your GitHub account:",
+  "email": "The email in which you can be contacted in:",
+  "general": "General section - comma separated values:",
+  "categories": "Categories - comma separated values:",
+  "useCases": "Use cases - comma separated values:",
+  "keywords": "Keywords - comma separated values:",
+  "dependencies": "Additional dependencies of the pack - comma separated values:"
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: no need to ask the user for dependencies, in init just initiate it as empty list

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "deprecated": "Whether pack is deprecated, options are True or False:",
+  "certification": "Certification of the pack, options are certified or verified :",
+  "serverMinVersion": "Server min version:",
+  "serverLicense": "Server license, options are standard or premium :",
+  "currentVersion": "Current version of the pack:",
+  "author": "Author of the pack:",
+  "url": "The url of support, should represent your GitHub account:",
+  "email": "The email in which you can be contacted in:",
+  "general": "General section - comma separated values:",
+  "categories": "Categories - comma separated values:",
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: for categories we should allow them to choose of the specific categories we have
Threat Intelligence and Data Enrichment, EDR, Network Security, etc

File: demisto_sdk/common/templates/PackData/metadata_template.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  "beta": "Whether is beta version, options are True or False :",
+  "deprecated": "Whether pack is deprecated, options are True or False:",
+  "certification": "Certification of the pack, options are certified or verified :",
+  "serverMinVersion": "Server min version:",
+  "serverLicense": "Server license, options are standard or premium :",
+  "currentVersion": "Current version of the pack:",
+  "author": "Author of the pack:",
+  "url": "The url of support, should represent your GitHub account:",
+  "email": "The email in which you can be contacted in:",
+  "general": "General section - comma separated values:",
```
Comment (anara123) [inline demisto_sdk/common/templates/PackData/metadata_template.json]: dont ask the user this

File: demisto_sdk/dev_tools/initiator.py [hunk 8]
```diff
@@ -137,6 +137,15 @@ def pack_init(self) -> bool:
 
         print_color(f"Successfully created the pack {self.dir_name} in: {self.full_output_path}", LOG_COLORS.GREEN)
 
+        with open(os.path.join(self.full_output_path, 'metadata.json'), 'a') as fp:
+            user_response = input("\nDo you want to fill pack's metadata file? Y/N ").lower()
+            fill_manually = True if user_response in ['y', 'yes'] else False
```
Comment (yaakovi) [inline demisto_sdk/dev_tools/initiator.py]:
```suggestion
            fill_manually = user_response in ('y', 'yes')
```

File: demisto_sdk/dev_tools/initiator.py [hunk 9]
```diff
@@ -145,6 +154,27 @@ def pack_init(self) -> bool:
...
+
+            for key, value in metadata_template.items():
+                value_from_user = input(f'\n{value} ')
+
+                if 'comma separated values' in value:
+                    value_from_user = [v.strip() for v in value_from_user.split(',')]
+
+                metadata_template[key] = value_from_user
+
+            if fill_manually:
```
Comment (yaakovi) [inline demisto_sdk/dev_tools/initiator.py]: remove condition, you only get here if `fill_manually` is true

File: demisto_sdk/common/templates/PackData/metadata_template.json (hunk 7 above)
Comment (IkaDemisto) [reply demisto_sdk/common/templates/PackData/metadata_template.json]: Should we remove it completely? Or init it as empty list in metadata?

File: demisto_sdk/common/templates/PackData/metadata_template.json (hunk 6 above)
Comment (IkaDemisto) [reply demisto_sdk/common/templates/PackData/metadata_template.json]: Do we have complete list of options? So I can add it as options?

File: demisto_sdk/common/templates/PackData/metadata_template.json (hunk 7 above)
Comment (IkaDemisto) [reply demisto_sdk/common/templates/PackData/metadata_template.json]: removed the general section for now

File: demisto_sdk/common/templates/PackData/metadata_template.json (hunk 6 above)
Comment (anara123) [reply demisto_sdk/common/templates/PackData/metadata_template.json]: https://github.com/demisto/content/blob/master/Tests/schemas/integration.yml#L30-L33

**Comment by IkaDemisto** [comment]
Removed template as we aren't using majority of it field to fill the `metadata.json`.  Building the template data dynamically.

File: demisto_sdk/commands/init/initiator.py [hunk 10]
```diff
@@ -146,6 +159,77 @@ def pack_init(self) -> bool:
...
+                                                                 option_message="\nPack category options: \n")]
+        tags = input("\nTags of the pack, comma separated values: ")
+        tags_list = [t.strip() for t in tags.split(',')]
+        metadata['tags'] = tags_list
+
+        return metadata
+
+    @staticmethod
+    def get_valid_user_input(options_list, option_message):
+        # category_options = "\nPack category options: \n"
```
Comment (yaakovi) [inline demisto_sdk/commands/init/initiator.py]: remove

**Pull request merged**